core/finality-grandpa/src/observer.rs

```diff
@@ -92,7 +92,7 @@ fn grandpa_observer<B, E, Block: BlockT<Hash=H256>, RA, S>(
 
 		// if the commit we've received targets a block lower than the last
 		// finalized, ignore it and continue with the current state
-		if commit.target_number < last_finalized_number {
+		if commit.target_number <= last_finalized_number {
```

Comment:
That looks like some different fix hiding in this pr. I think we should move this into a new pr + add a test for it.

Reply:
Yeah, I was just testing as the commit name was trying to suggest.

Reply:
Ahh okay :)